File: frontend/src/model/questionnaires/questionnaireQuestionOption.ts
export interface QuestionnaireQuestionOption {
  id: number;
  question_id: number;
  order: number;
  value: string;
  with_input: boolean;
}

export interface QuestionnaireQuestionOptionCreate {
  order: number;
  value: string;
  with_input: boolean;
}

export interface QuestionnaireQuestionOptionUpdate {
  id: number;
  question_id?: number;
  order?: number;
  value?: string;
  with_input: boolean;
}

File: learn-api/app/schemas/questionnaire_question_option.py
from pydantic import BaseModel

from typing import Optional


class QuestionnaireQuestionOptionBase(BaseModel):
    order: int
    value: str


class QuestionnaireQuestionOptionCreate(QuestionnaireQuestionOptionBase):
    question_id: int


class QuestionnaireQuestionOptionUpdate(QuestionnaireQuestionOptionBase):
    order: Optional[int]
    value: Optional[str]


class QuestionnaireQuestionOptionInDBBase(QuestionnaireQuestionOptionBase):
    id: int
    question_id: int

    class Config:
        orm_mode: True


class QuestionnaireQuestionOptionInDB(QuestionnaireQuestionOptionInDBBase):
    pass


class QuestionnaireQuestionOption(QuestionnaireQuestionOptionInDB):
    pass
